storage/libradb/src/ledger_store/ledger_info_test.rs

```diff
@@ -8,62 +8,82 @@ use libra_types::block_info::BlockInfo;
-                                 consensus_hash in any::<HashValue>(),
-                                 timestamp in any::<u64>()) -> (HashValue, HashValue, u64) {
-        (accu_hash, consensus_hash, timestamp)
-    }
-}
-
-prop_compose! {
-    fn arb_ledger_infos_with_sigs()(
-        partial_ledger_infos_with_sigs in vec(
-            any_with::<LedgerInfoWithSignatures>((1..3).into()).no_shrink(), 1..100
+fn arb_ledger_infos_with_sigs() -> impl Strategy<Value = Vec<(LedgerInfoWithSignatures, bool)>> {
+    (
+        vec(
+            (
+                any::<HashValue>(),
+                any::<HashValue>(),
+                any::<HashValue>(),
+                any::<u64>(),
+                0..10u64,
+                any::<bool>(),
+            ),
+            1..100,
         ),
-        start_version in 0..10000u64,
-    ) -> Vec<LedgerInfoWithSignatures> {
-        partial_ledger_infos_with_sigs
-            .iter()
-            .enumerate()
-            .map(|(i, p)| {
-                let ledger_info = p.ledger_info();
-                let version = start_version + i as u64;
-                let epoch: u64 = version / 3;
-                // epoch changes at versions that are multiplications of 3
-                let next_validator_set = if version % 3 == 2 {
-                    Some(ValidatorSet::new(vec![]))
-                } else {
-                    None
-                };
+        vec(0..100_000_000u64, 3),
+    )
+        .prop_map(|(block_params, mut ledger_params)| {
+            ledger_params.sort_unstable();
+            let mut epoch = ledger_params[0];
+            let mut round = ledger_params[1];
+            let mut version = ledger_params[2];
+            let mut last_new_epoch_version = u64::max_value();
 
-                LedgerInfoWithSignatures::new(
-                    LedgerInfo::new(
-                        BlockInfo::new(
-                            version,
-                            epoch,
-                            ledger_info.consensus_block_id(),
-                            ledger_info.transaction_accumulator_hash(),
-                            start_epoch + i as Version,
-                            ledger_info.timestamp_usecs(),
-                            next_validator_set,
-                        ),
-                        ledger_info.consensus_data_hash(),
-                    ),
-                    p.signatures().clone(),
+            block_params
+                .into_iter()
+                .map(
+                    |(
+                        block_id,
+                        accu_hash,
+                        consensus_data_hash,
+                        timestamp,
+                        block_size,
+                        mut is_new_epoch,
+                    )| {
+                        round += 1;
+                        version += block_size;
+                        is_new_epoch = is_new_epoch & (last_new_epoch_version == version);
```

Comment:
At the same version there will never be two epoch bumps, guaranteed, right? @zekun000

Reply:
yeah, you need a transaction to bump it